Optimized and mostly stable build

CarsonBurke opened the pull request.

CarsonBurke committed:
(dysfunctional) planning improvements

CarsonBurke committed:
Global clear commands

CarsonBurke committed:
Seeming fix to findClosestPos infinite recursion, spawn cancelling with killAllCreeps

CarsonBurke committed:
Visuals fix for findClosestPos

CarsonBurke committed:
Fix to findClosestPos localVisitedCoords

CarsonBurke committed:
Pixel selling improvements

CarsonBurke committed:
Minor fastFiller creep improvement

CarsonBurke committed:
minor builder spawning optimization

CarsonBurke committed:
Command improvements

CarsonBurke committed:
Implemented a max construction placement per room, mapVisuals room type fix

CarsonBurke committed:
Assumed fix to contruction cap not placing cSites

CarsonBurke committed:
Rampart planning accounting for roomDimensions fix

CarsonBurke committed:
Improvement to unprotected coord matrix value

pieterbrandsen committed:
Updated the dashboards

pieterbrandsen committed:
Added feature to use an different bot for each user

CarsonBurke committed:
internationalManager cache reset for some properties

CarsonBurke committed:
Fix and improvement to pixel selling, hubHauler new reservation system start

CarsonBurke committed:
improvement to pixel selling

Allorrian committed:
Corrected Pixel Selling for max profit, set buy/sell orders to be +/- 0.001 of the appropriate order, corrected order retrieval to not add completed orders to array

CarsonBurke committed:
Minor market improvements

CarsonBurke committed:
Merge branch 'Development' into Development

CarsonBurke committed:
Merge pull request #162 from Allorrian/Development

Corrected Pixel Selling for max profit, set buy/sell orders to be +/-  0.001 of the appropriate order, corrected order retrieval to not add completed orders to array

CarsonBurke committed:
Implemented optimized Allorrian market improvements

CarsonBurke committed:
Removal of inactive orders when reaching order limit

CarsonBurke committed:
Minor market improvement

CarsonBurke committed:
spawn request undefined maxCreeps in group spawning fix

Allorrian committed:
returned to order.reaminingAmount == 0 to prevent double orders

Appears orders are in an inactive state in the tick they are first placed. As such, when inactive orders are skipped when creating _myOrders, new orders are also skipped causing the bot to enter a second order.

Allorrian committed:
fixed max roomDimensions for exits in rampartPlanner

CarsonBurke committed:
Merge pull request #164 from Allorrian/Development

returned to order.reaminingAmount == 0 to prevent double orders

pieterbrandsen committed:
Updated data being sent to grafana

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
Modified safemode targets

CarsonBurke committed:
Significantly improved advancedRecycle

CarsonBurke committed:
advancedRecycle fixes and improvements

CarsonBurke committed:
advancedRecycle improvement, find closest in range function

CarsonBurke committed:
advancedRenew adjacent spawn finding fix

CarsonBurke committed:
Fix to melee defender under-spawning

CarsonBurke committed:
Very buggy hubHauler proper hubLink filling and factory usage

pieterbrandsen committed:
Added limit to stats

CarsonBurke committed:
source harvest position change, tower re-enable

CarsonBurke committed:
Significant hubHauler resource handling improvements

CarsonBurke committed:
hubHauler filling improvements and edge case catches

CarsonBurke committed:
hubHauler storing structures balancning fix

CarsonBurke committed:
Rampart road planning fix

CarsonBurke committed:
unneeded log removal

CarsonBurke committed:
Minor recycle inefficacy fix

CarsonBurke committed:
Limited scout count when RCL 8

CarsonBurke committed:
Fix to creeps trying to reserve enemy ramparted storing structures

CarsonBurke committed:
Fix to lastReset stat recording, start on asymmetric stamp planning

CarsonBurke committed:
(breaking) Minimized lab planning space usage

CarsonBurke committed:
Minor adjustment to lab protection range

CarsonBurke committed:
Further minimized lab stamp size usage, disabled planning debug visuals

pieterbrandsen committed:
Decreased auto pixel buying price

CarsonBurke committed:
Significant lab placement and general base planning improvements

CarsonBurke committed:
(breaking) Late fix to breaking version

CarsonBurke committed:
rampart planning on factory

pieterbrandsen committed:
Added an shard name to object

pieterbrandsen committed:
Added updated dashboard

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
Minor adjustment to breaking version reset order

CarsonBurke committed:
fix to scout advancing signing moveRequest error

CarsonBurke committed:
Minor defence spawning adjustments

CarsonBurke committed:
minor meleeDefender strength adjustment

CarsonBurke committed:
Fix to trying to spawn based on inactive storing structures

CarsonBurke committed:
Removal of maxRooms in advancedFindPath

pieterbrandsen committed:
Fixed bug in controllerLevel going to infinity

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

pieterbrandsen committed:
Adding logging

pieterbrandsen committed:
Fixed bug with extra char

pieterbrandsen committed:
Rows were expected to have datasources

pieterbrandsen committed:
Renamed attackRequestsManager to its own name

pieterbrandsen committed:
Removed extra room stats memory

pieterbrandsen committed:
Fixed bug where rooms were being requested when gcl was at max

Commit:
Handing there only being one source in a room.
Removing a few things that were printed to the console.
Adding code so that if one creep crashes, it doesn't stop the whole script.

CarsonBurke committed:
lab planning and centerUpgradePos planning fix, maintainer scaling fix

CarsonBurke committed:
(breaking) increasing breaking chance metric following previous commit

pieterbrandsen committed:
Updated server logging

Commit:
Fixing some suggestions from Panda

pieterbrandsen committed:
Added more detailed remote and per user stats

pieterbrandsen committed:
Added extra room to grafana

pieterbrandsen committed:
Fixed error, forgot one field for global stats

Commit:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
(breaking) Use of acronyms as common room memory keys

CarsonBurke committed:
Re-enabled endTickManager

CarsonBurke committed:
Simplified hauler reservation creation

Commit:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
Fix to GCLLeft / reservedGCL

Commit:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
Fix to reservedRCL accounting for commune count

Commit:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
Fix to responder when request is abandoned, Memory.communes -> global.communes, fastFiller container use after storage stoppage

CarsonBurke committed:
optional diagonal-usage planning optimizations

CarsonBurke committed:
(breaking change) extensions, observer and tower planning optimizations

CarsonBurke committed:
Deletion of non-neutral room claim requests

pieterbrandsen committed:
Removed users.js.example

pieterbrandsen committed:
Updated dashboard

pieterbrandsen committed:
Updated users

pieterbrandsen committed:
Merge branch 'Development' into Development

pieterbrandsen committed:
Added check for if steam api has an length of 0

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/plaidrabbit/The-International-Open-Source into Development

pieterbrandsen committed:
Fixed issue where steam_api_key want set to undefined

pieterbrandsen committed:
Added 2 1 source rooms to server

pieterbrandsen committed:
Error fix when controller was undefined (room without controller)

pieterbrandsen committed:
Removed logging of logs in performance server

pieterbrandsen committed:
Fixed roomStats controller level bug where it wasnt pushed

pieterbrandsen committed:
Updated stats dashboard

pieterbrandsen committed:
Merge pull request #179 from plaidrabbit/Development

Handing there only being one source in a room.

CarsonBurke committed:
hub size optimization

CarsonBurke committed:
Stop to builder spawning when enemy attackers are present

CarsonBurke committed:
Stop to fastFiller recycle attempts

CarsonBurke committed:
cardinalFlood for finding centerUpgradePos

CarsonBurke committed:
Tower repairing now recorded in room stats

CarsonBurke committed:
Reduced hauler carry part need calculations

CarsonBurke committed:
Inclusion of RCL 4 when checking for storage activity

CarsonBurke committed:
hubLink usage as receiver, fastFiller fill fix, hubLink definition fix

CarsonBurke committed:
Claimer accounts for move part count when pathing

CarsonBurke committed:
Inter room pathing optimizations

CarsonBurke committed:
removal of debug info

pieterbrandsen committed:
Removed error logging

pieterbrandsen committed:
Added energy sold/bought stats

pieterbrandsen committed:
Build error fix

pieterbrandsen committed:
Fixed problem with NaN stats

pieterbrandsen committed:
Added renew costs inside spawn cost stats

pieterbrandsen committed:
Made room controler level null when not my controller

pieterbrandsen committed:
Added extra check to get no errors when room is undefined

Commit:
Disable scout notifications going out.  They die a lot.
Fixing an issue with the reservation system.
Making a few changes so bad reservations don't cause the whole system to hang.
deleting bad claim requests if there's one present.

CarsonBurke committed:
base planning optimizations, hub source preference

CarsonBurke committed:
individual extension planning adjustment to follow rampart planning

CarsonBurke committed:
Road coord planning near exit fix

Commit:
Adding factory and lab functions

Commit:
Adding in a Commune Class, with the LabManager hanging off of it.

Commit:
cleanup

Commit:
more cleanup

CarsonBurke committed:
freeStore fix, creep class constructor check improvement, claim request management improvement, formatting

CarsonBurke committed:
Merge pull request #188 from plaidrabbit/Development

Small changes to improve stability/reduce noise

CarsonBurke committed:
Reservation optimizations, better spawn error checking

pieterbrandsen committed:
Added totalCreepCount instead of remoteCreepCount

CarsonBurke committed:
Better spawning debug info

pieterbrandsen committed:
Added respawn clear memory check

pieterbrandsen committed:
Gave credit

pieterbrandsen committed:
AveragedOverTick number was completly wrong

Commit:
more cleanup

Commit:
Merge remote-tracking branch 'origin/Development' into Development

Commit:
Fix for loading the Commune.

CarsonBurke committed:
Fix to NaN spawning cost

Commit:
Making the lab IDs not hard coded.
Removing a lot of the reservation code, to start removing it.  This way we're not modifiying the server objects.
Fixing where it wouldn't always pull energy when trying to fill mandatory energy items.

CarsonBurke committed:
configManager with migrations, reservationEfficacy, individual spawning bug fix

Commit:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

Commit:
More of getting rid of code that modifies the screeps server object's storage.

CarsonBurke committed:
remoteHauler spawning and need selection optimization, sourceHarvester and reserver pre-spawning optimization

CarsonBurke committed:
reseerver spawning fix, remote assigning more on sourceEfficacies

CarsonBurke committed:
Fix to fastFiller not withdrawing non-energy resources from containers

CarsonBurke committed:
Traffic optimizations plus priority by role, hauler size avg CPU account, builder spawning increase

Commit:
Merge remote-tracking branch 'origin/Development' into Development

CarsonBurke committed:
Traffic optimizations, fastFiller larger spawns, hauler size fix

CarsonBurke committed:
Stop to room visuals outside of allowed setting

CarsonBurke committed:
remote hauler spawning optimization

Commit:
Merge remote-tracking branch 'origin/Development' into Development

CarsonBurke committed:
(migration) defaultMinCacheAmount connected to CPU usage, max hauler size fix, pre ticks implemented into role files, recording of controller positions in memory and room getters

Commit:
Bunch of fixes.  Getting it to spawn another hauler a little earlier
some fixes around early game.
Making the temp default lab action G
Fixed a glitch in one of the factory conditions
Checking for some error/edge cases
adding stack traces into the error message
fixing a compiler error
doing a workaround for the freeCapacityAfterTick to replace reservations

CarsonBurke committed:
Data structure improvements, builder spawning improvement, construction site placing improvement

Commit:
Couple small fixes that were in my other branch

Commit:
Merge remote-tracking branch 'remotes/origin/Development' into Development

pieterbrandsen committed:
Added node 18.x

pieterbrandsen committed:
Moved energy input to level 1

plaidrabbit committed:
Merge pull request #195 from plaidrabbit/Development

Labs and Factory

pieterbrandsen committed:
Removed support for 18.x

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
upgrade position defining improvement

pieterbrandsen committed:
Changed delay in pushing private server stats

CarsonBurke committed:
Reversion of sub-optimal changes

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

pieterbrandsen committed:
Fixed bug in zero total creep count

pieterbrandsen committed:
Updated grafana schema and dashboard

CarsonBurke committed:
Traffic optimizations: no more irrational swapping

Commit:
Converting a lot of the creep code to using classes.

Commit:
Merge remote-tracking branch 'origin/Development' into Development

Commit:
Removing something from react?!?

plaidrabbit committed:
Merge pull request #204 from plaidrabbit/Development

Development

CarsonBurke committed:
Remote needs optimizations, base planning refactor start

CarsonBurke committed:
Implemented primitive remoteHauler relaying, structural improvements

CarsonBurke committed:
Relay improvement

CarsonBurke committed:
minor optimizations

CarsonBurke committed:
Significantly improved remote map visuals

CarsonBurke committed:
observer planning bug fix

Commit:
Adding in Lab code.

CarsonBurke committed:
Advanced lab protection

plaidrabbit committed:
Merge pull request #206 from plaidrabbit/Development

Adding in Lab code.

CarsonBurke committed:
Improved room manager class structures

pieterbrandsen committed:
Removed devcontainer

pieterbrandsen committed:
Updated server rooms

pieterbrandsen committed:
Updated serverStats

pieterbrandsen committed:
Cron sceduling bug fix

pieterbrandsen committed:
Increased retention

pieterbrandsen committed:
Added 2 users

CarsonBurke committed:
Hauling changes and other improvements

CarsonBurke committed:
Revert "Hauling changes and other improvements"

This reverts commit f4d69b9c5195114bcd94b8e662b5450d073ce522.

Commit:
MarketManager to class

Commit:
New market code, but it's disabled.  I left the existing market code in there as active.  But this way we can start working on it as a team.

pieterbrandsen committed:
Updated stats dashboard

pieterbrandsen committed:
Updated rooms to be used for shardVision

pieterbrandsen committed:
trackedRoom constant added

pieterbrandsen committed:
Wrong rooms set

Commit:
Small fix, so that it doesn't build roads if there's currently a structure in the way there.

Commit:
Delaying construction of ramparts until the structures are actually built.  ramparts for critical structures no longer have to be included in the base planning.
Adding in an error wrapper around the market manager
Adding in some guard code to ensure an exception doesn't happen if a room's memory is reset.

CarsonBurke committed:
(dysfunctional) attempt for remote harvesters and haulers source-based need response, spawning optimizations

maxsupermanhd committed:
Make log messages a bit smaller (they simply don't fit)

CarsonBurke committed:
Merge pull request #208 from maxsupermanhd/Development

Make log messages a bit smaller (they simply don't fit)

CarsonBurke committed:
Full transition to source-based remote spawning, relaying improvements, builder spawning optimizations

CarsonBurke committed:
fix to reserver under-spawning

Allorrian committed:
Updated Pixel Selling to get best BuyOrders and place sell orders during periods when the buy orders are below minPrice

CarsonBurke committed:
Relaying improvements, reserver spawning improvements, memHack organization

Allorrian committed:
Merge branch 'The-International-Screeps-Bot:Development' into Development

CarsonBurke committed:
Significant relay improvements

Allorrian committed:
Merge branch 'The-International-Screeps-Bot:Development' into Development

CarsonBurke committed:
Remote spawning optimizations

Commit:
Fixing a small edge case.

Commit:
Fixing build error.

CarsonBurke committed:
Relay recursion fixed, relay optimizations, remoteHarvester spawning improvements

Commit:
improving the typing of myCreeps

Commit:
D'oh.  Broke the build.

CarsonBurke committed:
Merge pull request #212 from Allorrian/Development

Updated Pixel Selling to use getBuyOrders() and place sell orders…

pieterbrandsen committed:
Removed types from main.ts

pieterbrandsen committed:
Updated PandaMaster code

pieterbrandsen committed:
Merge branch 'Development' into development-respawn

pieterbrandsen committed:
Merge pull request #218 from The-International-Screeps-Bot/development-respawn

When respawned, clear all Memory

Commit:
Working on the remote Hauling code.  It does a better job of spawning the correct haulers.

pieterbrandsen committed:
Added metrics processed to dashboard

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

Commit:
getting rid of the need to explicitly call updateNeeds

pieterbrandsen committed:
Renamed metrics for serviceInfo dashboard

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

Commit:
Improving remote hauling.

CarsonBurke committed:
Converted advancedFindPath to use an array of goals

CarsonBurke committed:
Merge bugs handling

CarsonBurke committed:
FIX Remote pathing in unsafe rooms, FIX import issue

CarsonBurke committed:
IMPROVEMENT minor claim request response optimizations

pieterbrandsen committed:
Updated serverStats dashboard timeframe

Commit:
Improvements towards adding boosting and remote performance.

Commit:
Ooops, broken build.

Commit:
Cleaning up the factory code some

Commit:
Fixing a bug where Remote Source Harvesters need to take into account if they are dieing.
Calculate the SE, adding in terrain costs (now swamps cost 5)
the priority for the remote haulers should be based on the correct SE.

pieterbrandsen reviewed (approved):
I approve, contact me on discord when going to merge because I would like to include my Discord bot stuff before merge so I dont get errors going forward on Master branch

pieterbrandsen committed:
Merge branch 'Development' of https://github.com/The-International-Screeps-Bot/The-International-Open-Source into Development

CarsonBurke committed:
Remote, room, commune management additions and improvements

FEATURE relaying while fatigued,
FEATURE remoteHauler pick up dropped energy below,
OPTIMIZATION reverted inefficient, remote-based spawning changes to remoteHauler spawning,
FEATURE class based roomVisuals,
BUG FIX proper functional order for communes,
FEATURE traffic options for creeps that will only yield to higher-priority ones
BUG FIX remote pre-spawning (besides remoteHauler)

CarsonBurke committed:
FIX mistake in avoidEnemyRanges

CarsonBurke committed:
Debug tools for remote needs issue

Commit:
Handing an edge case with the HH, for when the factory is over-filled with energy from breaking down batteries.

CarsonBurke committed:
FIX remoteHauler needs defining, OPTIMIZATION remoteHarvester transfer near full or to fill

CarsonBurke committed:
removal of remote debug tools

CarsonBurke committed:
Dynamic storage requirements for upgrader and terminal
